configuration/external_parameters.rst

```diff
@@ -537,6 +537,129 @@ Symfony provides the following env var processors:
+    Parses an absolute URL and returns its components.
+
+    .. code-block:: bash
+
+        # .env
+        MONGODB_URL="mongodb://db_user:db_password@127.0.0.1:27017/db_name"
+
+    .. configuration-block::
+
+        .. code-block:: yaml
+
+            # config/packages/mongodb.yaml
+            mongo_db_bundle:
+                clients:
+                    default:
+                        hosts:
+                        - { host: '%env(key:host:url:MONGODB_URL)%', port: '%env(key:port:url:MONGODB_URL)%' }
+                        username: '%env(key:user:url:MONGODB_URL)%'
+                        password: '%env(key:pass:url:MONGODB_URL)%'
+                connections:
+                    default:
+                        database_name: '%env(key:path:url:MONGODB_URL)%'
+
+        .. code-block:: xml
+
+            <!-- config/packages/mongodb.xml -->
+            <?xml version="1.0" encoding="UTF-8" ?>
+            <container xmlns="http://symfony.com/schema/dic/services"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://symfony.com/schema/dic/services
+                    http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+                <mongodb:config>
+                    <mongodb:client name="default" username="%env(key:user:url:MONGODB_URL)%" password="%env(key:pass:url:MONGODB_URL)%">
+                        <mongodb:host host="%env(key:host:url:MONGODB_URL)%" port="%env(key:port:url:MONGODB_URL)%"/>
+                    </mongodb:client>
+                    <mongodb:connections name="default" database_name="%env(key:path:url:MONGODB_URL)%"/>
+                </mongodb:config>
+            </container>
+
+        .. code-block:: php
+
+            // config/packages/mongodb.php
+            $container->loadFromExtension('mongodb', [
+                'clients' => [
+                    'default' => [
+                        'hosts' => [
+                            [
+                                'host' => '%env(key:host:url:MONGODB_URL)%',
+                                'port' => '%env(key:port:url:MONGODB_URL)%',
+                            ],
+                        ],
+                        'username' => '%env(key:user:url:MONGODB_URL)%',
+                        'password' => '%env(key:pass:url:MONGODB_URL)%',
+                    ],
+                ],
+                'connections' => [
+                    'default' => [
+                        'database_name' => '%env(key:path:url:MONGODB_URL)%',
+                    ],
+                ],
+            ]);
+
+    .. caution::
+
+        In order to ease extraction of the resource from the URL, The leading
+        ``/`` is trimed from the ``path`` component.
+
+    .. versionadded:: 4.3
+
+        The ``url`` processor was introduced in Symfony 4.3.
+
+``env(query_string:FOO)``
+    Parses an encoded string as if it were the query string passed via a URL.
+
+
+    .. code-block:: bash
```

Comment:
@javiereguiluz is bash ok or better terminal or text here for the code block?

Reply:
Note: this is the same code block used above in that file

Reply:
Thanks for the info, maybe we should change them all. Not sure

Reply:
For information purposes: we decided a few days ago to use `bash` for "bash things" (such as displaying the content of `.env` files) and use `terminal` for real consoles (when we show commands and their outputs).